shinylive/_main.py

```diff
@@ -469,6 +471,153 @@ def defunct_help(cmd: str) -> str:
 """
 
 
+# #############################################################################
+# ## shinylive.io url
+# #############################################################################
+
+
+@main.group(
+    short_help="Create or decode a shinylive.io URL.",
+    help="Create or decode a shinylive.io URL.",
+    no_args_is_help=True,
+    cls=OrderedGroup,
+)
+def url() -> None:
+    pass
+
+
+@url.command(
+    short_help="Create a shinylive.io URL from local files.",
+    help="""
+Create a shinylive.io URL for a Shiny app from local files.
+
+APP is the path to the primary Shiny app file.
+
+FILES are additional supporting files or directories for the app.
+
+On macOS, you can copy the URL to the clipboard with:
+
+    shinylive url encode app.py | pbcopy
+""",
+)
+@click.option(
+    "-m",
+    "--mode",
+    type=click.Choice(["editor", "app"]),
+    required=True,
+    default="editor",
+    help="The shinylive mode: include the editor or show only the app.",
+)
+@click.option(
+    "-l",
+    "--language",
+    type=click.Choice(["python", "py", "R", "r"]),
+    required=False,
+    default=None,
+    help="The primary language used to run the app, by default inferred from the app file.",
+)
+@click.option(
+    "-v", "--view", is_flag=True, default=False, help="Open the link in a browser."
+)
+@click.option(
+    "--no-header", is_flag=True, default=False, help="Hide the Shinylive header."
+)
+@click.argument("app", type=str, nargs=1, required=True, default="-")
+@click.argument("files", type=str, nargs=-1, required=False)
+def encode(
```

Comment:
A `--json` option would be useful for encoding.

Reply:
I'm curious how you see the JSON option being useful, but I completely agree that in making this possible we will definitely push to code towards a better structure.

Reply:
I've had a few occasions where I've used `shinylive export` and then made tweaks to the generated `app.json` file, and it might have been easier and quicker to be able to simply re-export the JSON from this command. But that is a pretty niche debugging use case, and not particularly important to do at this point.

Reply:
It's done :smile

Reply:
Wait, I think it's done. Right now it outputs the app bundle as JSON, i.e. taking the app and supporting file inputs and making their shinylive JSON representation. It occurred to me just now that maybe you'd want to take JSON inputs and return a shinylive URL. If that's the case, let me know, otherwise if you wanted JSON output we're all set.